modin/experimental/engines/omnisci_on_ray/frame/data.py

```diff
@@ -260,12 +260,34 @@ def groupby_agg(self, by, axis, agg, groupby_args, **kwargs):
             raise NotImplementedError("levels are not supported for groupby")
 
         groupby_cols = by_frame.columns.tolist()
-        agg_cols = [col for col in self.columns if col not in by_frame.columns]
+        if isinstance(agg, dict):
+            agg_cols = list(agg.keys())
+        else:
+            agg_cols = [col for col in self.columns if col not in by_frame.columns]
+            # If 'self' is Series and we're grouping on it
```

Comment:
Is it for a Series case only? Wouldn't group by all frame columns lead us here too?

Reply:
You're right, in that case, the result has to be empty. Gonna fix this case

Reply:
Roughly fixed that. However, there is still mismatching with pandas behavior for one-column frames (returning aggregated result instead of empty frame, looks not critical)